### PR TITLE
ethereum: Don't use cached block on `is_on_main_chain`

### PR DESCRIPTION
**Pull request by leoyvens**
As it turns out, the optimization in https://github.com/graphprotocol/graph-node/pull/1477 was incorrect wrt reverts.

The block stream relies on this function to check with the Ethereum node if a block is on the main chain, but it is assuming that the block being checked is final if it is past the reorg threshold. If the current subgraph head is in the cache when passed into `is_on_main_chain`, as long as it is present in the block cache it will be assumed to be final, which can lead to it not being reverted when it should be.

The most likely way for this bug to affect a subgraph is if a large reorg happens, such that by the time the last blocks are having the reorg processed by graph node they have fallen out of the threshold and are passed into `is_on_main_chain` and considered final. Since mainnet does not see large reorgs relative to the threshold this should not affect subgraphs currently on the network. In the hosted service, I have seen a Ropsten subgraph affected during a 200 block reorg, which was how I became aware of this bug.

The fix is to basically revert https://github.com/graphprotocol/graph-node/pull/1477.

### CODE REVIEW

File: server/http/tests/server.rs [hunk 1]
```diff
@@ -107,7 +107,7 @@ mod test {
                         // Send an empty JSON POST request
                         let client = Client::new();
                         let request =
-                            Request::post(format!("http://localhost:8001/subgraphs/id/{}", id))
+                            Request::post(format!("http://localhost:8007/subgraphs/id/{}", id))
```
Comment (leoyvens) [inline server/http/tests/server.rs:110]: This is unrelated, just picked a port that would stop conflicting with graph-node defaults.

File: chain/ethereum/src/ethereum_adapter.rs [hunk 2]
```diff
@@ -1123,71 +1118,29 @@ impl EthereumAdapterTrait for EthereumAdapter {
...
-        };
-        let num_hashes = hashes.len();
-        let logger1 = logger.clone();
-        let confirm_block_hash = move |hash: &Option<H256>| {
-            // If there was more than one hash, now that we know what the
-            // 'right' one is, get rid of all the others
-            if let Some(hash) = hash {
-                if block_is_final && num_hashes > 1 {
-                    chain_store
-                        .confirm_block_hash(block_number, hash)
```
Comment (lutter) [inline chain/ethereum/src/ethereum_adapter.rs:1144]: It's probably not a big deal in practice, but by never calling this, we'll hang on to forked blocks forever; duplicate blocks IIRC can lead to trouble resolving block numbers to block hashes. I don't think it's a big deal, since we don't make changes for nonfinal blocks anyway, but something to keep in mind.

**Review by lutter — APPROVED** [review]

File: chain/ethereum/src/ethereum_adapter.rs (hunk 2 above)
Comment (leoyvens) [reply chain/ethereum/src/ethereum_adapter.rs:1144]: I kept the `fn confirm_block_hash` around, in case we can find a different spot for it. But I suppose an incoming firehose integration means we won't have to think to hard about this.
